src/Microsoft.Health.Dicom.Core/Extensions/DicomDatasetExtensions.cs

```diff
@@ -158,5 +149,71 @@ public static void AddValueIfNotNull<T>(this DicomDataset dicomDataset, DicomTag
                 dicomDataset.Add(dicomTag, value);
             }
         }
+
+        public static IDictionary<IndexTag, DicomTag> GetDicomTags(this DicomDataset dicomDataset, IEnumerable<IndexTag> indexTags)
```

Comment:
add summary, add unit tests

Reply:
unit tests:
1.  when contains both standtag and private tags, both are processed correctly

Reply:
Unit test:
2. If standard tag not exist in dataset, should not return
3. If private tag not exist in dataset, should not return (VR not match, tagpath not exist)